File: 442-find-all-duplicates-in-an-array/find-all-duplicates-in-an-array.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var findDuplicates = function(nums) {

    let freq = new Array(100001).fill(0);
    let arr = new Array();
    for(let i = 0; i<nums.length; i++)
    {
        freq[nums[i]]++;
    }
    for(let i = 0; i<freq.length; i++)
    {
        if(freq[i] > 1)
        {
            arr.push(i);
        }
    }
    return arr;
};

File: 442-find-all-duplicates-in-an-array/find-all-duplicates-in-an-array.java
class Solution {
    public List<Integer> findDuplicates(int[] nums) {
        
        int freq [] = new int[100001];
         List<Integer> result = new ArrayList<>();

        for(int i = 0; i<nums.length; i++)
        {
            freq[nums[i]]++;
        }
        for(int i = 0; i<freq.length; i++)
        {
            if(freq[i] > 1)
            {
                result.add(i);
            }
        }
        return result;
    }
}